src/Servers/IIS/AspNetCoreModuleV2/RequestHandlerLib/filewatcher.cpp

```diff
@@ -261,16 +297,94 @@ HRESULT
         }
     }
 
-    if (fFileChanged && !_lStopMonitorCalled)
+    if (fAppOfflineChanged && !_lStopMonitorCalled)
     {
         // Reference application before
         _pApplication->ReferenceApplication();
         RETURN_LAST_ERROR_IF(!QueueUserWorkItem(RunNotificationCallback, _pApplication.get(), WT_EXECUTEDEFAULT));
     }
 
+    if (fDllChanged && m_fShadowCopyEnabled && !_lStopMonitorCalled)
+    {
+        // Reset timer for dll checks
+        LOG_INFO(L"Detected dll change, resetting timer callback which will eventually trigger shutdown.");
+        m_Timer.CancelTimer();
+        m_Timer.InitializeTimer(FILE_WATCHER::TimerCallback, this, 5000, 3000);
+    }
+
     return S_OK;
 }
 
+
+VOID
+CALLBACK
+FILE_WATCHER::TimerCallback(
+    _In_ PTP_CALLBACK_INSTANCE Instance,
+    _In_ PVOID Context,
+    _In_ PTP_TIMER Timer
+)
+{
+    Instance;
```

Comment:
😄 I thought there was some `UNUSED` macro or something for these cases.

Reply:
I think I can just remove the variable names, seems to be the more common way for unused.